datafusion/optimizer/src/push_down_filter.rs

```diff
@@ -511,25 +511,20 @@ impl OptimizerRule for PushDownFilter {
         "push_down_filter"
     }
 
+    fn apply_order(&self) -> Option<ApplyOrder> {
+        Some(ApplyOrder::TopDown)
+    }
+
     fn try_optimize(
         &self,
         plan: &LogicalPlan,
-        config: &dyn OptimizerConfig,
+        _config: &dyn OptimizerConfig,
```

Comment:
Will it lose information if we disable `config` here?

Reply:
> Will it lose information if we disable `config` here?

I didn't disable it, just because it's only used recursively, so it needs to be prefixed with `_`.

It's hint by `cargo clippy`

Reply:
> just because it's only used recursively

In the original code, we pass the `config` to the `optimize_children`, but now, it is just a placeholder for self recursion, which means we never use the `config` anymore.